registration/authorize.go

```diff
@@ -180,6 +180,14 @@ func authorizeNodeCommon(
 	// Save the node information into storage if not skipped
 	if !opts.WithSkipStorage {
 		if err := nodeInfo.Store(ctx, storage, opt...); err != nil {
+			// If using a storage implementation where Store does not overwrite existing records, detect
+			// and handle receiving duplicate reqInfo by returning the stored nodeInfo
+			if errors.As(err, &types.DuplicateRecordError{}) {
+				loadNodeInfo, err := types.LoadNodeInformation(ctx, storage, nodeInfo.Id)
+				if err == nil {
+					return loadNodeInfo, nil
+				}
```

Comment:
I'd add an `err = fmt.Errorf("(%s) error loading node information after duplicate record detected: %w", op, err)` or something similar so that the error chain is a bit more informative about why the error is coming from the load function